cupy/core/core.pyx

```diff
@@ -1565,6 +1565,37 @@ cdef class ndarray:
     # TODO(okuta): Implement __setslice__
     # TODO(okuta): Implement __contains__
 
+    # numpy/ufunc compat
+    def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
+
+        """Apply unary or binary ufunc to this array
+
+        If binary, only allow if second argument is another cupy ndarray or
+        a number, i.e., raise ValueError instead of silently converting a
+        numpy array.
+        """
+        import cupy  # top-level ufuncs
+        if 'out' in kwargs:
+            # need to unfold tuple argument in kwargs
```

Comment:
Cloud you check tuple length?

Reply:
As far as I understand, the out tuple could in theory have more entries, for some future ufunc that has multiple outputs. I don't think numpy as any of these, never mind cupy - so I agree that an exception could be raised if the tuple length>1. This is what numpy does, e.g., `np.sin(x, out=(x, x)) -> ValueError`.

Reply:
OK, I understood.